tests/xgboost/test_xgboost_autolog.py

```diff
@@ -454,3 +461,42 @@ def test_xgb_autolog_does_not_break_dmatrix_serialization(bst_params, tmpdir):
 
     dataset.save_binary(tmpdir.join("dataset_serialization_test"))  # serialization should not throw
     xgb.DMatrix(tmpdir.join("dataset_serialization_test"))  # deserialization also should not throw
+
+
+@pytest.mark.large
+def test_lgb_autolog_configuration_options(bst_params):
+    iris = datasets.load_iris()
+    X = pd.DataFrame(iris.data[:, :2], columns=iris.feature_names[:2])
+    y = iris.target
+
+    with mlflow.start_run() as run:
+        mlflow.xgboost.autolog(log_input_example=False, log_model_signature=False)
+        dataset = xgb.DMatrix(X, y)
+        xgb.train(bst_params, dataset)
+    model_conf = get_model_conf(run.info.artifact_uri)
+    assert "saved_input_example_info" not in model_conf.to_dict()
+    assert "signature" not in model_conf.to_dict()
```

Comment:
We can use `pytest.mark.parametrize` here.

```suggestion
@pytest.mark.large
@pytest.mark.parametrize("log_input_example", [True, False])
@pytest.mark.parametrize("log_model_signature", [True, False])
def test_lgb_autolog_configuration_options(bst_params):
    iris = datasets.load_iris()
    X = pd.DataFrame(iris.data[:, :2], columns=iris.feature_names[:2])
    y = iris.target
    
    assert ("saved_input_example_info" in model_conf.to_dict()) == log_input_example
    assert ("signature" in model_conf.to_dict()) == log_model_signature
```

Reply:
With `pytest.mark.parametrize`, we can easily understand that this test verifies `mlflow.xgboost.autolog` works as expected for each `(log_input_example, log_model_signature)` combination.

Reply:
thats awesome!

Reply:
thanks Haru, this makes it very clean! I love it :D